src/DotNetty.Transport/Channels/DefaultChannelId.cs

```diff
@@ -0,0 +1,419 @@
+                pId = Process.GetCurrentProcess().Id;
+            }
+            catch
+            {
+                // ignored
+            }
+
+            if (pId < 0)
+            {
+                pId = ThreadLocalRandom.Value.Next(MaxProcessId + 1);
+            }
+            return pId;
+        }
+
+        public static DefaultChannelId NewInstance()
+        {
+            var id = new DefaultChannelId();
+            id.Init();
+
+            return id;
+        }
+
+        static byte[] DefaultMachineId()
+        {
+            // Find the best MAC address available.
+            byte[] notFound = { byte.MaxValue };
+            byte[] bestMacAddr = notFound;
+            IPAddress bestIpAddr = IPAddress.Loopback;
+            var ifaces = new SortedDictionary<NetworkInterface, IPAddress>();
+            try
+            {
+                foreach (NetworkInterface iface in NetworkInterface.GetAllNetworkInterfaces())
+                {
+                    var addrs = iface.GetIPProperties().UnicastAddresses;
+                    var addr = addrs.FirstOrDefault(a => !IPAddress.IsLoopback(a.Address));
+                    if (addr != null)
+                    {
+                        ifaces.Add(iface, addr.Address);
+                    }
+                }
+            }
+                //catch (SocketException e)
+                //{
+                //    logger.warn("Failed to retrieve the list of available network interfaces", e);
+                //}
+            catch
+            {
+                // ignored
+            }
+
+            foreach (var entry in ifaces)
+            {
+                NetworkInterface iface = entry.Key;
+                IPAddress addr = entry.Value;
+                //todo check if the iface is virtual(there is no equivvalent method in .Net like in java)
+                byte[] macAddr = iface.GetPhysicalAddress().GetAddressBytes();
+                bool replace = false;
+                int res = CompareAddresses(bestMacAddr, macAddr);
+                if (res < 0)
+                {
+                    replace = true;
+                }
+                else if (res == 0)
+                {
+                    res = CompareAddresses(bestIpAddr, addr);
+                    if (res < 0)
+                    {
+                        replace = true;
+                    }
+                    else if (res == 0)
+                    {
+                        if (bestMacAddr.Length < macAddr.Length)
+                        {
+                            replace = true;
+                        }
+                    }
+                }
+
+                if (replace)
+                {
+                    bestMacAddr = macAddr;
+                    bestIpAddr = addr;
+                }
+            }
+
+            if (bestMacAddr == notFound)
+            {
+                bestMacAddr = new byte[MachineIdLen];
+                ThreadLocalRandom.Value.NextBytes(bestMacAddr);
```

Comment:
`Logger.Warn`. pls go through orig code to see where logging should be returned and either add it (as its available now in dev) or leave TODOs.